tensorflow_serving/batching/batching_session.cc

```diff
@@ -340,19 +464,59 @@ Status BatchingSession::MergeInputTensors(
 
   // For each input tensor name, a vector of tensors from the individual tasks.
   std::map<string, std::vector<Tensor>> tensors_to_merge;
-
```

Comment:
This is a lot of code to in-line in this method. It would be nice to preserve the non-padding code if possible. Would it be possible to make a surgical AddPadding() helper method/function?

Also, can you add a "bool pad_variable_length_inputs" or something along those lines to BatchingSessionOptions.

I'm envisioning something that looks like:
... do some stuff
if (options_.pad_variable_length_inputs) {
  AddPadding(...);
}
... do more stuff

Is that doable?